modules/internal/ChapelArray.chpl

```diff
@@ -2027,7 +2030,28 @@ module ChapelArray {
     proc tail(): this._value.eltType {
       return this[this.domain.high];
     }
-
+    pragma "no doc"
+    proc resizeAllocDomain(d: domain, d2: domain, factor=arrayAsVecGrowthFactor, direction=1, grow=1) {
+      const lo = d.low,
+            hi = d.high,
+            size = hi - lo + 1;
+      var newSize: int;
+      if grow > 0 {
+        newSize = max(size+1, (size*factor):int); // Always grow by at least 1.
```

Comment:
Doesn't have to be done for this commit, but I think this should use `chpl_mem_good_alloc_size()` to reduce slack. All this really means is that the size will be rounded up to the actual size the allocator will use

https://github.com/facebook/folly/blob/master/folly/docs/FBVector.md#the-jemalloc-connection is a good section on why using `chpl_mem_good_alloc_size()` makes sense and that doc is really good general info on implementing a fast vector.